src/DataStructures/DataBox.hpp

```diff
@@ -232,52 +235,52 @@ class TaggedDeferredTuple
     template <class Tuple, class... Ts>
     static constexpr bool enable_explicit() noexcept {
       return not tuples_detail::all<
-          std::is_convertible<Ts, item_type<Tags>>::value...>::value;
+          std::is_convertible<Ts, Deferred<item_type<Tags>>>::value...>::value;
     }
 
     template <class Tuple, class... Ts>
     static constexpr bool enable_implicit() noexcept {
       return tuples_detail::all<
-          std::is_convertible<Ts, item_type<Tags>>::value...>::value;
+          std::is_convertible<Ts, Deferred<item_type<Tags>>>::value...>::value;
     }
   };
 
   template <class Dummy>
   struct tuple_like_constructor<true, true, Dummy> {
     template <class Tuple, class... Ts>
     static constexpr bool enable_explicit() noexcept {
-      return not tuples_detail::all<
-                 std::is_convertible<Ts, item_type<Tags>>::value...>::value and
+      return not tuples_detail::all<std::is_convertible<
+                 Ts, Deferred<item_type<Tags>>>::value...>::value and
              (not tuples_detail::all<std::is_convertible<
-                  Tuple, item_type<Tags>>::value...>::value and
+                  Tuple, Deferred<item_type<Tags>>>::value...>::value and
               not tuples_detail::all<std::is_constructible<
-                  item_type<Tags>, Tuple>::value...>::value and
-              not tuples_detail::all<
-                  std::is_same<item_type<Tags>, Ts>::value...>::value);
+                  Deferred<item_type<Tags>>, Tuple>::value...>::value and
+              not tuples_detail::all<std::is_same<Deferred<item_type<Tags>>,
+                                                  Ts>::value...>::value);
```

Comment:
is_same_v

Reply:
I'm keeping the TaggedTuple and TaggedDeferredTuple portions the same as their [standalone versions](https://github.com/nilsdeppe/tuples), which means C++11 compliant

Reply:
okay, but this in in DataBox.hpp

Reply:
Sure, that's where TaggedDeferredTuple lives, though. Do you want me to move that to a separate file?

Reply:
yes, if you want to keep certain things c++11 compliant, I think it makes sense to keep them separate

Reply:
I would rather make DataBox C++11 compatible, but regardless of that, TaggedDeferredTuple needs to be associated with DataBox, so the natural place to factor it out to would `src/DataStructures/TaggedDeferredTuple.hpp`. Is that okay?